code/controllers/subsystem/weather.dm

```diff
@@ -15,10 +15,21 @@ var/datum/subsystem/weather/SSweather
 /datum/subsystem/weather/fire(resumed = FALSE)
 	if(flags & SS_NO_FIRE)
 		return
-	if(map.climate)
-		var/datum/climate/C = map.climate
-		C.tick()
+	if(climates.len)
+		for(var/datum/climate/C in climates)
+			C.tick()
 	else
 		flags |= SS_NO_FIRE
 		pause()
 		message_admins("Weather subsystem was paused due to lack of climate.")
+
+/datum/subsystem/weather/proc/get_climate(var/z)
+	for(var/datum/climate/C in climates)
```

Comment:
as a failsafe, I would have this return climates[1] if there are no other entries

Reply:
+1